edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,131 @@
+use futures::executor::block_on;
+use futures::join;
+use nix::sys::signal::{self, Signal};
+use nix::unistd::Pid;
+use signal_hook::{iterator::Signals, SIGINT, SIGTERM};
+use std::io::Error;
+use std::process::{exit, Child, Command, Stdio};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use tracing::{error, info, subscriber, Level};
+use tracing_subscriber::fmt::Subscriber;
+
+fn main() -> Result<(), Error> {
+    block_on(async_main())
+}
+
+async fn async_main() -> Result<(), Error> {
+    init_logging();
+    info!("Starting watchdog");
+
+    let should_shutdown = match register_sigterm_listener() {
+        Ok(should_shutdown) => should_shutdown,
+        Err(e) => {
+            error!("Failed to register sigterm listener. Shutting down.");
+            return Err(e);
+        }
+    };
+
+    // start edgehub and blow up if can't start
+    let mut edgehub = Command::new("dotnet")
+        .arg("/app/Microsoft.Azure.Devices.Edge.Hub.Service.dll")
+        .stdout(Stdio::inherit())
+        .spawn()
+        .expect("Failed to start Edge Hub process");
+    info!("Launched Edge Hub process with pid {:?}", edgehub.id());
+
+    // unwrap broker if started, else shutdown edgehub and exit
+    let maybe_broker = match Command::new("/usr/local/bin/mqttd")
```

Comment:
is this var name ok?

Reply:
it's not an Option, so `maybe_` is not very applicable here.
you can do a bit different:
```rust
let broker = Command::new("/usr/local/bin/mqttd")
        .stdout(Stdio::inherit())
        .spawn() ;
match broker 
{
  Ok(broker )  => broker,
  Err(e) => { warn!(...); exit(1); }
}